wordpress-mobile/WordPress-iOS
# Site Domains: Modernize UI 1/2

Partially implements #22262
Figma rvVgXupiyONUcJMgPSoMFj-fi-3518_13838

Modernize Site Domains UI and re-use elements from All Domains.

Second PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/22311, I think it's the best to test on second PR

## Issue

All Domains uses `/all-domains/` endpoint while Site Domains uses `sites/%d/domains` endpoint which produces a bit different data:

- Most notably, `sites/%d/domains` does not include `status` (Active/Verifying/Expired..) which is shown on All Domains view. Determining `status` involves quite a lot of logic (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Syvo%2Sqbznvaf%2Sqbznva%2Qfgnghf%2Qerfbyire%2Sqbznva%2Qfgnghf%2Qerfbyire.cuc%3Se%3Qsn1n97n3%23ertvfgrerq_qbznva_fgnghf-og) therefore we cannot and shouldn't do it on the frontend site.
- All Domains entities include `site_slug` which is needed to open domain management.

I decided to only focus on UI updates in the scope of this PR, which means that I'm not including the endpoint changes to include "Status" and opening of domain management. I will do it in another PR (https://github.com/wordpress-mobile/WordPress-iOS/pull/22311).

## Solution

1. Create `SiteDomainsViewModel` and move data loading logic there
2. Build free domain, and other domain sections, reusing `AllDomainsListCardView.ViewModel`
3. Create `PrimaryDomainView` badge
4. Update `SiteDomainsView` to use a new ViewMode and reuse `AllDomainsListCardView`
5. To have consistent spacing between cards I created rows that look like `InsetGroupStyle` instead of using this style directly, since spacing is only supported from iOS17

<img src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/a39ee083-5ead-4371-99b8-3a58c3e04178" alt="Simulator Screenshot - iPhone 15 - 2023-12-27 at 10 07 39" width="250">

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/46f835d4-a7ff-43f3-b067-7b3d060ab930

## To test:

Can be tested together with https://github.com/wordpress-mobile/WordPress-iOS/pull/22311

1. Check `Menu -> Domains` UI:
- With an account that has a domain (domains)
- With an account that only has a free domain

2. Confirm UI updates after purchasing the domain
3. Confirm UI appears correctly in dark mode
4. Confirm UI appears correctly with increased font sizes
5. Confirm UI appears correctly in a horizontal mode

## Regression Notes
1. Potential unintended areas of impact

Breaking existing SiteDomains behavior

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


UI Changes testing checklist:
